File: Easy Guys/Finding 3-Digit Even Numbers/solution.py
from collections import Counter

class Solution:
    def findEvenNumbers(self, digits: List[int]) -> List[int]:
        count = Counter()
        for digit in digits:
            count[digit] += 1
        
        result = []
        for i in range(1, 10):
            for j in range(0, 10):
                for k in range(0, 10, 2):
                    needI = 1
                    needJ = 1
                    needK = 1
                    if j == i:
                        needJ += 1
                    if k == j:
                        needK += 1
                    if k == i:
                        needK += 1
                    if count[i] >= needI and count[j] >= needJ and count[k] >= needK:
                        result.append(i*100 + j*10 + k)
        return result
                        

File: Easy Guys/Finding 3-Digit Even Numbers/solution.cpp
class Solution {
public:
    vector<int> findEvenNumbers(vector<int>& digits) {
        vector<int> counter(10);
        for(int d: digits) counter[d]++;
        vector<int> result;
        for(int d1 = 1; d1 <= 9; ++d1) {
            if(!counter[d1]) continue;
            counter[d1]--;
            for(int d2 = 0; d2 <= 9; ++d2) {
                if(!counter[d2]) continue;
                counter[d2]--;
                for(int d3 = 0; d3 <= 9; d3 += 2) {
                    if(!counter[d3]) continue;
                    result.emplace_back(d1 * 100 + d2 * 10 + d3);
                }
                counter[d2]++;
            }
            counter[d1]++;
        }
        return result;
    }
};
